lib/sidekiq/cli.rb

```diff
@@ -269,6 +269,19 @@ def validate!
         logger.info @parser
         die(1)
       end
+
+      errors = []
+      [:concurrency, :index, :timeout].each do |arg|
```

Comment:
`index` is not a field you would pass in the config file (since it's process-specific) so you don't need to validate it.  0 is not a legal value for concurrency and timeout so you can do this:

```ruby
raise ArgumentError, "Invalid timeout value #{options[:timeout]} in #{options[:config_file]}" if options.has_key?(:timeout) && options[:timeout].to_i == 0
raise ArgumentError, "Invalid concurrency value #{options[:concurrency]} in #{options[:config_file]}" if options.has_key?(:concurrency) && options[:concurrency].to_i == 0
```